### PR TITLE
Fixed bar chart color display. Specified variable types. #1462

### PR DESCRIPTION
**Pull request by KD1712**
Fixes bar chart color display. Sorting the displaying graphData, graphColorIndicators and conditionsFilterOptions(for correct display order). Gave variable types previously designated as 'any'.

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts [hunk 1]
```diff
@@ -203,16 +217,24 @@ export class EnrollmentOverTimeComponent implements OnChanges, OnInit, OnDestroy
...
+      // Sort the series array alphabetically based on the condition names.
+      // This ensures consistent ordering of conditions in the resulting graph.
+      series.sort((a, b) => a.name.localeCompare(b.name));
+
       return {
         name:
-          this.selectedDateFilter === DATE_RANGE.LAST_SEVEN_DAYS
-            ? this.dateToString(new Date(graphData.date), days)
-            : months[new Date(graphData.date).getMonth()],
+        this.selectedDateFilter === DATE_RANGE.LAST_SEVEN_DAYS
```
Comment (RidhamShah) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:227]: Indentation is not right. (Prettier errors)

**Review by Yagnik56 — COMMENTED** [review]
The code looks good, you added 2 blank lines and uneven space around ` = ` with few changes. Can you add a prettier extension, it will help you fix this.

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts [hunk 2]
```diff
@@ -37,6 +37,19 @@ export {
   IExperimentEnrollmentDetailStats,
   DATE_RANGE,
 };
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:39]: add a blank line

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts [hunk 3]
```diff
@@ -37,6 +37,19 @@ export {
...
+export interface ExperimentConditionFilterOptions {
+  code: string;
+  id: string
+}
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:43]: add a blank line

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts [hunk 4]
```diff
@@ -37,6 +37,19 @@ export {
...
+export interface ExperimentPartitionFilterOptions {
+  id:string
+  point:string
+  twoCharacterId:string
+}
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:48]: add a blank line

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.html [hunk 5]
```diff
@@ -116,7 +116,7 @@
     <div class="chart">
       <div class="chart-data">
         <ngx-charts-bar-vertical-stacked
-          [scheme]="colorScheme"
+          [scheme] = "colorScheme"
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.html:119]: remove this spaces around `=` for consistency with other params

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts [hunk 6]
```diff
@@ -44,6 +47,7 @@ export class EnrollmentOverTimeComponent implements OnChanges, OnInit, OnDestroy
   copyGraphData: IEnrollmentStatByDate[] = [];
   isInitialLoad = true;
   showLabelOfxAxis = true;
+  graphColorIndicators = [];
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:50]: Can you please add the datatype for this?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts [hunk 7]
```diff
@@ -44,6 +47,7 @@ export class EnrollmentOverTimeComponent implements OnChanges, OnInit, OnDestroy
   copyGraphData: IEnrollmentStatByDate[] = [];
   isInitialLoad = true;
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:48]: Can you please add the boolean datatype for this?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts [hunk 8]
```diff
@@ -44,6 +47,7 @@ export class EnrollmentOverTimeComponent implements OnChanges, OnInit, OnDestroy
...
   showLabelOfxAxis = true;
```
Comment (ppratikcr7) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:49]: Can you please add the boolean datatype for this?

**Review by ppratikcr7 — CHANGES_REQUESTED** [review]
Please check small changes requested.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts [hunk 9]
```diff
@@ -91,7 +96,9 @@ export class EnrollmentOverTimeComponent implements OnChanges, OnInit, OnDestroy
       this.experiment.conditions.forEach((condition) => {
         this.conditionsFilterOptions.push({ code: condition.conditionCode, id: condition.id });
         this.selectedCondition.push(condition.id);
+        this.conditionsFilterOptions.sort((a, b) => a.code.localeCompare(b.code));
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:99]: The sort is done in every loop. This statement can be moved outside of the loop

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts (hunk 1 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts]: Fixed indentation with Prettier

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts (hunk 4 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:50]: Added blank lines in Model

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.html (hunk 5 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.html]: Removed this whitespace for consistent code

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts (hunk 6 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts]: Assigned datatype to this.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts (hunk 8 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts:50]: It gave  @typescript-eslint/no-inferrable-types error. So boolean type not assigned.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts (hunk 9 above)
Comment (KD1712) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/enrollment-over-time/enrollment-over-time.component.ts]: Moved sorting outside the loop

**Review by Yagnik56 — APPROVED** [review]
everything looks good to me.

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**